provider/datagen/src/options.rs

```diff
@@ -0,0 +1,85 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::prelude::*;
+use icu_provider::prelude::*;
+use std::collections::{HashSet, BTreeSet};
+
+#[non_exhaustive]
+#[derive(serde::Serialize, serde::Deserialize)]
+pub struct Options {
+    #[serde(with = "data_key_as_str")]
+    pub keys: HashSet<DataKey>,
```

Comment:
It would be good to have an enum to allow more ways to specify keys like you did with locales. Only one for now but we can have a way to add more. For example, all keys for some component.